pkg/interactor/save_data_test.go

```diff
@@ -0,0 +1,89 @@
+package interactor
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/CESARBR/knot-cloud-storage/pkg/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	// ErrSaveDataDB is returned when the database fails to save data
+	ErrSaveDataDB = errors.New("error when trying to save data")
+)
+
+type SaveDataTestCase struct {
+	name             string
+	authorization    string
+	deviceID         string
+	payload          []entities.Payload
+	data             entities.Data
+	fakeLogger       *mocks.FakeLogger
+	fakeDataStore    *mocks.FakeDataStore
+	fakeThingService *mocks.FakeThingService
+	expectedError    error
+}
+
+var sdCases = []SaveDataTestCase{
+	{
+		"authorization token not provided",
+		"",
+		"",
+		nil,
+		entities.Data{},
+		&mocks.FakeLogger{},
+		&mocks.FakeDataStore{},
+		&mocks.FakeThingService{ReturnErr: errors.New("no access token provided")},
+		errors.New("no access token provided"),
+	},
+	{
+		"data successfully saved on the database",
+		"authorization-token",
+		"fc3fcf912d0c290a",
+		[]entities.Payload{{SensorID: 1, Value: 999}},
+		entities.Data{From: "fc3fcf912d0c290a", Payload: entities.Payload{SensorID: 1, Value: 999}, Timestamp: time.Now()},
+		&mocks.FakeLogger{},
+		&mocks.FakeDataStore{},
+		&mocks.FakeThingService{},
+		nil,
+	},
+	{
+		"failed to save the data on the database",
+		"authorization-token",
+		"fc3fcf912d0c290a",
+		[]entities.Payload{{SensorID: 1, Value: 999}},
+		entities.Data{From: "fc3fcf912d0c290a", Payload: entities.Payload{SensorID: 1, Value: 999}, Timestamp: time.Now()},
```

Comment:
Notice `time.Now()` will give us a different value every time we run it, this will probably cause tests not passing. Could you fix it? Some people faced the same problem when testing things that involve `time`, as you can see [here](https://stackoverflow.com/questions/18970265/is-there-an-easy-way-to-stub-out-time-now-globally-during-test), [here](https://www.reddit.com/r/golang/comments/640vz3/what_approach_do_you_use_to_mock_timenow/) and [here](https://smartystreets.com/blog/2015/09/go-testing-part-5-testing-with-time/).

Reply:
Yeah, of course. Actually I'm working on the solution. As soon as I finish it, I'll notify you.

Reply:
Done!